doc/faq.rst

```diff
@@ -378,6 +378,100 @@ password fields give no feedback, not even whether a character has been typed.
 Try using the number row instead, toggling Num Lock key, or SSHing into the
 virtual machine.
 
+SSH stops :code:`ch-image` build with interactive queries
+---------------------------------------------------------
+
+This often occurs during an SSH-based Git clone. For example::
+
+  $ cat ./Dockerfile
+  FROM alpine:latest
+  RUN apk add git openssh
+  RUN git clone git@github.com:hpc/charliecloud.git
+  $ ch-image build -t foo -f ./Dockerfile .
+  [...]
+  3 RUN ['/bin/sh', '-c', 'git clone git@github.com:hpc/charliecloud.git']
+  Cloning into 'charliecloud'...
+  The authenticity of host 'github.com (140.82.113.3)' can't be established.
+  RSA key fingerprint is SHA256:nThbg6kXUpJWGl7E1IGOCspRomTxdCARLviKw6E5SY8.
+  Are you sure you want to continue connecting (yes/no/[fingerprint])?
+
+At this point, the build stops while SSH waits for input.
+
+This happens even if you have :code:`github.com` in your
+:code:`~/.ssh/known_hosts`. This file is not available to the build because
+:code:`ch-image` runs :code:`ch-run` with :code:`--no-home`, so :code:`RUN`
+instructions can't see anything in your home directory.
+
+Solutions include:
+
+  1. Approve the connection interactively by typing :code:`y`. Note this will
+     record details of the connection within the image, including IP address
+     and the fingerprint.
+
+  2. Turn off host key checking in the image's system `SSH config
+     <https://man.openbsd.org/ssh_config>`_. There are multiple ways to do
+     this, but one is::
+
+       $ cat ./Dockerfile
+       FROM alpine:latest
+       RUN apk add git openssh
+       RUN printf 'StrictHostKeyChecking=no\nUserKnownHostsFile=/dev/null\n' \
+           >> /etc/ssh/ssh_config
+       RUN git clone git@github.com:hpc/charliecloud.git
+       $ ch-image build -t foo -f ./Dockerfile .
+       [...]
+         5 RUN ['/bin/sh', '-c', 'git clone git@github.com:hpc/charliecloud.git']
+       Cloning into 'charliecloud'...
+       Warning: Permanently added 'github.com' (RSA) to the list of known hosts.
+       remote: Enumerating objects: 10337, done.
+       remote: Counting objects: 100% (601/601), done.
+       remote: Compressing objects: 100% (299/299), done.
+       remote: Total 10337 (delta 400), reused 441 (delta 297), pack-reused 9736
+       Receiving objects: 100% (10337/10337), 12.77 MiB | 22.31 MiB/s, done.
+       Resolving deltas: 100% (7061/7061), done.
+       grown in 4 instructions: foo
+
+     Check your institutional policy on whether this is permissible, though
+     it's worth noting that users `almost never
+     <https://www.usenix.org/system/files/login/articles/105484-Gutmann.pdf>`_
+     verify the host fingerprints anyway.
+
+     This will not record details of the connection in the image.
+
+     More complex edit commands will be needed if either of these directives
```

Comment:
As you note, automated editing of the SSH config can get hairy.  Instead of recommending that, I'd replace this 2nd solution with setting `${GIT_SSH}` to something like `"ssh -o 'StrictHostKeyChecking=no' -o 'UserKnownHostsFile=/dev/null'"` to start off with.  You can always mention in a note that adding those settings to an appropriate `Host foo` block in either the system or the user SSH config file is a more robust way of accomplishing the same thing, but this way the detailed solution is something that will "always just work" as opposed to something potentially error-prone.